rust/core-lib/src/tabs.rs

```diff
@@ -491,6 +496,27 @@ impl<W: Write + Send + 'static> Documents<W> {
                 self.plugins.stop_plugin(&view_id, &plugin_name);
                 None
             }
+            PluginRpc  { view_id, receiver, mut rpc } => {
+                // inserting view_id into params here because the plugin needs
+                // to know where this command originates. This feels a bit
+                // clumbsy though. At the very least we should use some obscure
+                // key so we don't collide with any possible user-supplied keys?
+                // Or we could validate keys on plugin load or something.
```

Comment:
I think it would make more sense to have the view the command is run on be an argument to the custom RPC function with a `View` argument type. The plugin would have to explicitly add a `view` argument to the command if it needed the view, like how Rust methods require an explicit `self`. Then that argument would be implicitly filled in by the front end without it presenting a UI for it, although it could for example in a parameter filling dialog show that argument with the view file name the command is being run on.

This allows a distinction between for example global, buffer and view commands in a very functional-programming-esque way. It conveys if the command actually needs a view, allowing it to be potentially be called programmatically in contexts that don't have a view, and paving the way for other implicit argument types later.